src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java

```diff
@@ -219,6 +225,23 @@ public final void apply(Project project) {
         // (but that's internal)
         project.getPluginManager().apply("java-base");
 
+        // Create "platform" configuration in root project, which will hold the strictConstraints
+        NamedDomainObjectProvider<Configuration> gcvLocksConfiguration = project.getConfigurations()
+                .register("gcvLocks", conf -> {
+                    conf.getAttributes().attribute(Usage.USAGE_ATTRIBUTE, internalUsage);
+                    conf.getOutgoing().capability(GCV_LOCKS_CAPABILITY);
+                    conf.setCanBeResolved(false);
+                    conf.setVisible(false);
+                });
+
+        ProjectDependency locksDependency =
```

Comment:
why do we create a self dependency?

Reply:
it's not actually "self" but pointing to a particular configuration (indirectly).
This is to pick up the locks configuration which is defined in the root project.